File: 그리디/leetcode45(Medium)-Jump Game II.js
// 66ms(88.5%), 44MB(66.9%)
/**
 * @param {number[]} nums
 * @return {number}
 */
var jump = function (nums) {
  let answer = 0
  let curFar = 0
  let curEnd = 0

  for (let index = 0; index < nums.length - 1; index += 1) {
    curFar = Math.max(curFar, index + nums[index])

    if (index === curEnd) {
      answer += 1
      curEnd = curFar
    }
  }

  return answer
}

File: 그리디/leetcode45(Medium)-Jump Game II.py
# 136ms(68.89%), 15.1MB(54.98%)
class Solution:
    def jump(self, nums: List[int]) -> int:
        answer = 0
        cur_far = cur_end = 0

        for index in range(len(nums) - 1):
            cur_far = max(cur_far, index + nums[index])

            if index == cur_end:
                answer += 1
                cur_end = cur_far

        return answer